migration/migration.go

```diff
@@ -0,0 +1,68 @@
+package migration
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/NethermindEth/juno/db"
+)
+
+type revision func(db.DB) error
+
+var revisions = []revision{
+	revision0000,
+}
+
+func MigrateIfNeeded(targetDB db.DB) error {
+	version, err := SchemaVersion(targetDB)
+	if err != nil {
+		return err
+	}
+
+	for i := version; i < uint64(len(revisions)); i++ {
+		if err = revisions[i](targetDB); err != nil {
+			return err
+		}
+
+		// revision returned with no errors, bump the version
+		if err = targetDB.Update(func(txn db.Transaction) error {
+			var versionBytes [8]byte
+			binary.BigEndian.PutUint64(versionBytes[:], i+1)
+			return txn.Set(db.SchemaVersion.Key(), versionBytes[:])
+		}); err != nil {
+			return err
+		}
```

Comment:
Schema version of the targetDB determines which set of revisions need to be applied to the database. After revision is successfully executed, which may update the database, schema version is incremented by 1.

For example;

Assume an empty DB, which has a schema version of `0`. So we start applying the revisions from the very first one in the list and increment the schema version as we go. After the loop exists we end up with a database which's schema version is `len(revisions)+1`.

After running that Juno version for a while, if the user updates it's Juno version which adds new revisions to the list, `MigrateIfNeeded` will skip the already applied revisions and only apply the new ones. It will be able to do this since the schema version it reads from the database will be non-zero and that is what we use to initialize the `i` loop variable.

Reply:
Please add this explanation as comments